src/System.Text.Utf8/src/System/Text/Utf8/Utf8CodeUnit.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace System.Text.Utf8
+{
+    public struct Utf8CodeUnit : IEquatable<Utf8CodeUnit>
```

Comment:
Here's the thing: Whenever we release Utf8String we will need a way of providing how to assign a single character to Utf8String. If you decide to use byte, you won't be able to implement assignment as byte is not a character, same for char, as char is not UTF8 character and thus can be multibyte. There is no standard type representing it. If you decide to use Utf8CodeUnit you can easily add compiler support for ascii characters or "surrogate" utf8 string characters. Byte seems inappropriate in this context.